File: 나라티바(NOVA)/Narrativa_Backend/src/main/java/com/nova/narrativa/domain/notice/entity/Notice.java
package com.nova.narrativa.domain.notice.entity;

import com.nova.narrativa.domain.admin.entity.AdminUser;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(
        name = "notices",
        indexes = {
                @Index(name = "idx_notice_admin_user_id", columnList = "admin_user_id"),
                @Index(name = "idx_notice_status", columnList = "status"),
                @Index(name = "idx_notice_created_at", columnList = "createdAt")
        }
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Notice {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String title;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Status status = Status.ACTIVE;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "admin_user_id", nullable = false)
    private AdminUser createdBy;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    public enum Status {
        ACTIVE,
        INACTIVE
    }
}

File: 나라티바(NOVA)/Narrativa_Admin/src/types/notice.ts
// src/types/notice.ts

// Notice Status Enum
export type NoticeStatus = 'ACTIVE' | 'INACTIVE';

// Base Notice Type (공통 필드)
export interface NoticeBase {
  title: string;
  content: string;
  status: NoticeStatus;
}

// Notice Type (전체 Notice 타입)
export interface Notice extends NoticeBase {
  id: number;
  createdBy: String;
  createdAt: string;
  updatedAt: string;
}

// Create Notice Input Type (생성 시 필요한 필드)
export interface NoticeCreate extends NoticeBase {
  createdBy: String;
}

// Update Notice Input Type (수정 시 필요한 필드)
export interface NoticeUpdate {
  id: number;
  title?: string;
  content?: string;
  status?: NoticeStatus;
}

// Notice List Response Type (목록 조회 응답 타입)
export interface NoticeListResponse {
  notices: Notice[];
  totalPages: number;
  currentPage: number;
  totalItems: number;
}